pkg/controllers/work/apply_controller.go

```diff
@@ -375,41 +391,64 @@ func (r *ApplyWorkReconciler) applyUnstructured(ctx context.Context, gvr schema.
 		return nil, ManifestNoChangeAction, err
 	}
 
+	klog.InfoS("generation for curObj", "generation", curObj.GetGeneration())
+
 	// check if the existing manifest is managed by the work
 	if !isManifestManagedByWork(curObj.GetOwnerReferences()) {
 		err = fmt.Errorf("resource is not managed by the work controller")
 		klog.ErrorS(err, "skip applying a not managed manifest", "gvr", gvr, "obj", manifestRef)
 		return nil, ManifestNoChangeAction, err
 	}
 
+	klog.InfoS("Before trying to migrate")
+	// Migrate old objects with annotations mnifest hash and last modified annotation to use configmap instead, will be removed after migration is complete.
+	if err := r.migrateToConfigMap(ctx, gvr, curObj, owner); err != nil {
+		klog.ErrorS(err, "cannot migrate manifest to use config map", "manifest", manifestObj.GetName(), "namespace", manifestObj.GetNamespace())
+		return nil, ManifestNoChangeAction, err
+	}
+
+	// get config map or create it if it doesn't exist, need to set curObj generation to config map when creating it again
+	curObjConfigMap, err := r.getConfigMap(ctx, curObj, owner)
+	if err != nil {
+		return nil, ManifestNoChangeAction, err
+	}
+
+	klog.InfoS("generation for configmap", "generation", curObjConfigMap.GetGeneration())
+
+	// set config map name annotation on manifest object to prevent deletion during patch.
+	setConfigMapNameAnnotation(curObjConfigMap.GetName(), manifestObj)
+
+	// Use manifest generation to check if configmap is outdated
+	if curObj.GetGeneration() != curObjConfigMap.GetGeneration() {
```

Comment:
Hi Arvind! A few issues here:

* first of all, IIRC Kubernetes API server does not track generations for `configMaps`; the field will always be 0
* This checks compares the generations of two very different objects, which will only work if the two values are in-sync (there are no guarantees)

Reply:
Yes, was checking the logs and realized the generation for config map is always zero and the manifest's generation is also zero after patch